subsys/dfu/src/dfu_ctx_mcuboot.c

```diff
@@ -11,15 +11,22 @@
 
 LOG_MODULE_REGISTER(dfu_ctx_mcuboot, CONFIG_DFU_CTX_LOG_LEVEL);
 static struct flash_img_context flash_img;
+static size_t offset;
 
 int dfu_ctx_mcuboot_init(void)
 {
+	static bool initialized = false;
 
-	int err = flash_img_init(&flash_img);
+	if (!initialized) {
+		int err = flash_img_init(&flash_img);
```

Comment:
Why don't we de-initialize this when we call done?

Also reset `initialized` on done if we can do so.

Except for this everything looks fine